### PR TITLE
Mapreduce default parameter tuning - sort & copy

### PR DESCRIPTION
**Pull request by krisss85**
This is the proposal to increase mapreduce sort buffer to reduce number of spills and improve performance. Furthermore increase number of parallel copies cut time required to copy the data from mappers to reducers by allocating more threads for this operation.

### CODE REVIEW

File: spydra/src/main/resources/dataproc_defaults.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
...
+      "properties": "mapred:mapreduce.jobhistory.done-dir=gs://${LOG_BUCKET}/history/${CLIENT_ID}/done,mapred:mapreduce.jobhistory.intermediate-done-dir=gs://${LOG_BUCKET}/history/${CLIENT_ID}/${UUID}/done_intermediate,mapred:mapreduce.task.io.sort.mb=600,mapred:mapreduce.reduce.shuffle.parallelcopies=20,mapred:mapreduce.jobhistory.move.interval-ms=1000,mapred:mapreduce.output.fileoutputformat.compress=true,mapred:mapreduce.map.output.compress=true,mapred:mapreduce.output.fileoutputformat.compress.type=BLOCK,core:fs.gs.parent.timestamp.update.substrings.includes=gs://${LOG_BUCKET}/history,yarn:yarn.log-aggregation-enable=true,yarn:yarn.nodemanager.remote-app-log-dir=gs://${LOG_BUCKET}/logs/${CLIENT_ID},yarn:yarn.log-aggregation.retain-seconds=-1,core:fs.gs.implicit.directory.repair=false"
```
Comment (varjoranta) [inline spydra/src/main/resources/dataproc_defaults.json]: Could you add a short section into README regarding this change, i.e. just shortly saying what would be the minimum supported instance size to be used in Dataproc?

Comment (krisss85) [reply spydra/src/main/resources/dataproc_defaults.json]: I made an extra comment in ##### Submission Gotchas
We may also want to refactor this long properties string. Move the parameters to some Map object?

Comment (varjoranta) [reply spydra/src/main/resources/dataproc_defaults.json]: I'm happy with how it is now. I wouldn't start adding code complexity for this just yet. Getting close though.

**Comment by krisss85** [comment]
As agreed it is too specific to the way we use Spydra, so let's keep in minimum viable product here as anyone can fine tune the parameters to their own need.

**Pull request closed without merging**